extensions/vault/runtime/src/main/java/io/quarkus/vault/config/VaultServerRuntimeConfig.java

```diff
@@ -0,0 +1,172 @@
+package io.quarkus.vault.config;
+
+import static io.quarkus.vault.internal.LogConfidentialityLevel.LOW;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.vault.internal.LogConfidentialityLevel;
+
+@ConfigRoot(name = "vault", phase = ConfigPhase.RUN_TIME)
+public class VaultServerRuntimeConfig {
+
+    public static final String DEFAULT_K8S_TOKEN_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/token";
+    public static final String DEFAULT_KV_SECRET_ENGINE_PATH = "secret";
+    public static final String KV_SECRET_ENGINE_VERSION_V1 = "1";
+    public static final String KV_SECRET_ENGINE_VERSION_V2 = "2";
+    public static final String DEFAULT_RENEW_GRACE_PERIOD = "1H";
+    public static final String DEFAULT_CACHE_PERIOD = "10M";
+    public static final String K8S_CACERT = "/var/run/secrets/kubernetes.io/service-account/ca.crt";
+    public static final String DEFAULT_CONNECT_TIMEOUT = "5S";
+    public static final String DEFAULT_READ_TIMEOUT = "1S";
+    public static final String DEFAULT_TLS_SKIP_VERIFY = "false";
+
+    /**
+     * vault url. eg https://localhost:8200
+     */
+    @ConfigItem
+    public Optional<String> url;
+
+    /**
+     * k8s auth role
+     */
+    @ConfigItem
+    public Optional<String> role;
+
+    /**
+     * authentication type when logging in to get a client token. possible values:
+     * - k8s: kubernetes authentication as defined in https://www.vaultproject.io/api/auth/kubernetes/index.html
+     * - userpass (through properties userpassUsername and userpassPassword)
```

Comment:
It should be valid Javadoc or Asciidoctor as it will be integrated in the documentation.

If you want to use Asciidoctor, see: https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/extension-authors-guide.adoc#3-configuration-reference-documentation

Reply:
done